packages/strapi-admin/admin/src/containers/Webhooks/ListView/tests/reducer.test.js

```diff
@@ -0,0 +1,205 @@
+    const webhookToDelete = 1;
+    const action = {
+      type: 'SET_WEBHOOKS_TO_DELETE',
+      id: webhookToDelete,
+      value: false,
+    };
+
+    const expectedState = state.set(
+      'webhooksToDelete',
+      state.get('webhooksToDelete').remove(webhookToDelete)
+    );
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should update webhooks and empty webhooksToDelete on WEBHOOKS_DELETED', () => {
+    const webhooks = [
+      {
+        id: 1,
+        name: 'webhook 1',
+        url: 'http://localhost:5000',
+        headers: {},
+        events: ['entry.create', 'entry.update', 'entry.delete'],
+        isEnabled: true,
+      },
+      {
+        id: 2,
+        name: 'webhook 2',
+        url: 'http://localhost:4000',
+        headers: {},
+        events: ['media.create', 'media.update'],
+        isEnabled: false,
+      },
+    ];
+    const updatedWebhooks = [
+      {
+        id: 2,
+        name: 'webhook 2',
+        url: 'http://localhost:4000',
+        headers: {},
+        events: ['media.create', 'media.update'],
+        isEnabled: false,
+      },
+    ];
+
+    const state = initialState
+      .set('webhooksToDelete', [1])
+      .set('webhooks', fromJS(webhooks));
+    const action = {
+      type: 'WEBHOOKS_DELETED',
+    };
+
+    const expectedState = state
+      .set('webhooks', fromJS(updatedWebhooks))
+      .set('webhooksToDelete', []);
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should update webhooks and empty webhookToDelete on WEBHOOK_DELETED', () => {
+    const webhooks = [
+      {
+        id: 3,
+        name: 'webhook 1',
+        url: 'http://localhost:5000',
+        headers: {},
+        events: ['entry.create', 'entry.update', 'entry.delete'],
+        isEnabled: true,
+      },
+      {
+        id: 4,
+        name: 'webhook 2',
+        url: 'http://localhost:4000',
+        headers: {},
+        events: ['media.create', 'media.update'],
+        isEnabled: false,
+      },
+      {
+        id: 5,
+        name: 'webhook 2',
+        url: 'http://localhost:4000',
+        headers: {},
+        events: ['media.create', 'media.update'],
+        isEnabled: false,
+      },
+    ];
+    const updatedWebhooks = [
+      {
+        id: 3,
+        name: 'webhook 1',
+        url: 'http://localhost:5000',
+        headers: {},
+        events: ['entry.create', 'entry.update', 'entry.delete'],
+        isEnabled: true,
+      },
+      {
+        id: 5,
+        name: 'webhook 2',
+        url: 'http://localhost:4000',
+        headers: {},
+        events: ['media.create', 'media.update'],
+        isEnabled: false,
+      },
+    ];
+
+    const webhookIdToDelete = 4;
+
+    const state = initialState
+      .set('webhookToDelete', webhookIdToDelete)
+      .set('webhooks', fromJS(webhooks));
+
+    const action = {
+      type: 'WEBHOOK_DELETED',
+      index: webhooks.findIndex(webhook => webhook.id === webhookIdToDelete),
```

Comment:
Same here there is too much logic, if this function is tested somewhere it is ok but here you should definitely hardcode the index.